src/components/SubscribeModal.jsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import { Button, Modal, Form } from "semantic-ui-react";
+import {
+  CardNumberElement,
+  injectStripe,
+  CardCVCElement,
+  CardExpiryElement,
+} from "react-stripe-elements";
+import axios from "axios";
+
+function SubscribeModal({ onSubscribe, stripe }) {
+  const [open, setOpen] = React.useState(false);
+
+  const onCancelHandler = (event) => {
+    event.preventDefault();
+    setOpen(false);
+  };
+
+  const performPayment = async (stripeToken) => {
+    let headers = JSON.parse(localStorage.getItem("credentials"));
+    let response = await axios.post(
+      "/subscriptions",
+      { stripeToken: stripeToken },
+      { headers: headers }
```

Comment:
Why are you passing the headers differently here from previous API calls. Consistency is key, no way is right or wrong but choose one and stick to that

Reply:
Noted for this. Thanks.